x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapper.java

```diff
@@ -497,11 +498,38 @@ private static Query createConcatenationQuery(RegExp r) {
 
         }
 
+        private static Query createCharacterClassQuery(RegExp r) {
+            List<Query> queries = new ArrayList<>();
+            int maxClauseCount = 0;
+            for (int i = 0; i < r.from.length; i++) {
+                // TODO: consider expanding this to allow for character ranges as well (need additional tests and performance eval)
+                if (r.from[i] == r.to[i]) {
+                    maxClauseCount += r.to[i] - r.from[i];
+                    if (maxClauseCount > MAX_CLAUSES_IN_APPROXIMATION_QUERY) {
+                        return new MatchAllDocsQuery();
+                    }
+                    for (int j = r.from[i]; j <= r.to[i]; j++) {
+                        String cs = Character.toString(j);
+                        String normalizedChar = toLowerCase(cs);
+                        queries.add(new TermQuery(new Term("", normalizedChar)));
+                    }
```

Comment:
I think we can simplify this quite a bit, ` maxClauseCount += r.to[i] - r.from[i]` is always 0 now and the loop is just iterating one value so it can go too. Maybe we can add as protection outside of the main loop:

```
                    if (r.from.length > MAX_CLAUSES_IN_APPROXIMATION_QUERY) {
                        return new MatchAllDocsQuery();
                    }
```